### PR TITLE
Make favicon configurable

### LINKED ISSUE #18: Make favicon configurable
Please make custom favicon configurable with site params

### PR DESCRIPTION
**Pull request by RPGillespie6**
Fix #18 

Favicon can now be configured with, e.g.:

```
[params]
favicon = "favicon.ico"
```

### CODE REVIEW

File: layouts/partials/head.html [hunk 1]
```diff
@@ -16,8 +16,12 @@
 
 <!-- Icons -->
 <link rel="apple-touch-icon-precomposed" sizes="144x144" href="{{ "img/apple-touch-icon-144-precomposed.png" | absURL }}">
+{{ if isset .Site.Params "favicon" }}
+<link rel="shortcut icon" href="{{ .Site.BaseURL }}{{ .Site.Params.favicon }}">
```
Comment (panr) [inline layouts/partials/head.html]:
```suggestion
<link rel="shortcut icon" href="{{ .Site.Params.favicon | absURL }}">
```

File: layouts/partials/head.html [hunk 2]
```diff
@@ -16,8 +16,12 @@
...
+{{ else }}
 {{ with $.Site.Params.ThemeColor }}
```
Comment (panr) [inline layouts/partials/head.html]:
```suggestion
{{ else }}
  <link rel="shortcut icon" href="{{ printf "img/favicon/%s.png" $.Site.Params.ThemeColor | absURL }}">
{{ end }}
```

If we have `if/else` we don't need `with`. Lets go context free here ;-) (line 23 in sugestion is also very important)

File: layouts/partials/head.html [hunk 3]
```diff
@@ -16,8 +16,12 @@
 
 <!-- Icons -->
 <link rel="apple-touch-icon-precomposed" sizes="144x144" href="{{ "img/apple-touch-icon-144-precomposed.png" | absURL }}">
+{{ if isset .Site.Params "favicon" }}
```
Comment (panr) [inline layouts/partials/head.html]:
```suggestion
{{ if isset $.Site.Params "favicon" }}
```

**Comment by RPGillespie6** [comment]
I think that's everything. Let me know if the README needs to be beefed up more.

**Comment by panr** [comment]
We're good to go! Thanks! 🎉

**Pull request merged**